packages/arb-token-bridge-ui/src/hooks/useError.ts

```diff
@@ -0,0 +1,171 @@
+import { useCallback } from 'react'
+import * as Sentry from '@sentry/react'
+import { isUserRejectedError } from '../util/isUserRejectedError'
+import { useNetworks } from './useNetworks'
+
+/**
+ * Categories for classifying errors
+ */
+export type ErrorCategory =
+  | 'token_validation'
+  | 'token_approval'
+  | 'token_transfer'
+  | 'contract_interaction'
+  | 'contract_validation'
+  | 'claim'
+  | 'allowance_check'
+  | 'network_request'
+  | 'network_response'
+  | 'transaction_preparation'
+  | 'transaction_execution'
+  | 'bridge_validation'
+  | 'bridge_operation'
+  | 'user_input'
+  | 'user_rejection'
+  | 'system'
+  | 'unknown'
+
+/**
+ * Parameters for the `handleError` function
+ */
+export interface HandleErrorParams {
+  /** The original error object caught. */
+  error: unknown
+  /** A specific, unique identifier for the *operation* or context being attempted (e.g., 'cctp_approve_token', 'eth_deposit'). */
+  label: string
+  /** Optional: Caller-determined category for Sentry tagging. */
+  category?: ErrorCategory | string
```

Comment:
Any reason we add string? If we need something else than `ErrorCategory`, I would add the change to `ErrorCategory` no?